pkg/pluginmanager/manager.go

```diff
@@ -737,23 +742,49 @@ func installOrUpgradePlugin(p *discovery.Discovered, version string, installTest
 	}
 
 	// Log message based on different installation conditions
-	logPluginInstallationMessage(p, version, plugin != nil, isPluginAlreadyInstalled)
+	installingMsg, installedMsg, errMsg := getPluginInstallationMessage(p, version, plugin != nil, isPluginAlreadyInstalled)
 
-	if plugin == nil {
-		binary, err := fetchAndVerifyPlugin(p, version)
-		if err != nil {
-			return err
+	var spinnerErr, pluginErr error
+	var sow component.OutputWriterSpinner
+
+	// Initialize the spinner if the spinner is allowed
+	if component.IsTTYEnabled() {
+		// Set spinner options
+		spinnerOptions := component.OutputWriterSpinnerOptions{
+			SpinnerOptions: []component.OutputWriterSpinnerOption{
+				component.WithSpinnerFinalMsg(installedMsg, log.LogTypeINFO),
+			},
+		}
+		// Initialize the spinner
+		sow, spinnerErr = component.NewOutputWriterSpinnerWithSpinnerOptions(os.Stderr, component.TableOutputType, installingMsg, true, spinnerOptions)
+		if spinnerErr != nil {
+			log.Info(installingMsg)
+		}
+		if sow != nil {
+			defer sow.StopSpinner()
 		}
+	}
 
-		plugin, err = installAndDescribePlugin(p, version, binary)
-		if err != nil {
-			return err
+	var binary []byte
+	if plugin == nil {
+		binary, pluginErr = fetchAndVerifyPlugin(p, version)
+		if pluginErr == nil {
+			plugin, pluginErr = installAndDescribePlugin(p, version, binary)
 		}
 	}
-	if installTestPlugin {
-		if err := doInstallTestPlugin(p, plugin.InstallationPath, version); err != nil {
-			return err
+	if pluginErr == nil && installTestPlugin {
+		pluginErr = doInstallTestPlugin(p, plugin.InstallationPath, version)
+	}
+
+	if pluginErr != nil {
+		if sow != nil {
+			sow.SetFinalText(fmt.Sprintf("%s, error info:%s", errMsg, pluginErr.Error()), log.LogTypeERROR)
 		}
+		return pluginErr
+	}
+	// Stop the spinner before updating the plugin info and initializing the plugin
+	if sow != nil {
+		sow.StopSpinner()
```

Comment:
Should we remove this one since we have the `defer sow.StopSpinner()` above which will always get called?

Reply:
We don't need the spinner any more from this point as installation done, that why we are stopping here.